---
layout: workshop      # DON'T CHANGE THIS.
root: .               # DON'T CHANGE THIS.
venue: "RNAseq analysis in R"        # brief name of host site without address (e.g., "Euphoric State University")
address: "LAB-14, 700 Swanston Street, Carlton VICTORIA 3053"      # street address of workshop (e.g., "123 Forth Street, Blimingen, Euphoria")
country: "Australian"      # country (lowercase two-letter ISO code such as "fr" - see https://en.wikipedia.org/wiki/ISO_3166-1)
language: "en"     # language (lowercase two-letter ISO code such as "fr" - see https://en.wikipedia.org/wiki/ISO_639-1)
latlng: "-37.800229, 144.964256"       # fractional latitude and longitude (e.g., "41.7901128,-87.6007318"; you can use http://www.latlong.net/)
humandate: "May 11-12, 2016"    # human-readable date (e.g., "Feb 17-18, 2020")
humantime: "9:00 am - 4:30 pm"    # human-readable time (e.g., "9:00 am - 4:30 pm")
startdate: 2016-05-11      # use YYYY-MM-DD format like 2015-01-01
enddate: 2016-05-12        # use YYYY-MM-DD format like 2015-01-02
instructor: ["Belinda Phipson", "Maria Doyle", "Jovana Maksimovic", "Harriet Dashnow", "Matt Ritchie"] # list of names like ["Kay McNulty", "Betty Jennings", "Betty Snyder"]
helper: ["Jane Hawkey", "Sepideh Foroutan", "Ramyar Molania", "Chenkai Ma", "Luke Zappia", "Katrina Bell", "Anthony Hawkins", "Enid Lam", "Yih-Chih Chan", "Matt Wakefield", "Piyush Madhamshettiwar"]     # list of names like ["Marlyn Wescoff", "Fran Bilas", "Ruth Lichterman"]
contact: "combine@combine.org.au"      # contact email address for workshop organizer, such as "grace@hopper.org"
etherpad: http://pad.software-carpentry.org/2016-05-11-RNAseq
eventbrite:       # optional (insert the alphanumeric key for Eventbrite registration, e.g., "1234567890AB")
---


<!-- See instructions in the comments below for how to edit specific sections of this workshop template. -->



<!--
  HEADER

  Edit the values in the block above to be appropriate for your workshop.
  If the value is not 'true', 'false', 'null', or a number, please use
  double quotation marks around the value, unless specified otherwise.
  And run 'tools/check' *before* committing to make sure that changes are good.
-->

<!--
  EVENTBRITE

  This block includes the Eventbrite registration widget if
  'eventbrite' has been set in the header.  You can delete it if you
  are not using Eventbrite, or leave it in, since it will not be
  displayed if the 'eventbrite' field in the header is not set.
-->
{% if page.eventbrite %}
<iframe
  src="https://www.eventbrite.com/tickets-external?eid={{page.eventbrite}}&ref=etckt"
  frameborder="0"
  width="100%"
  height="248px"
  scrolling="auto">
</iframe>
{% endif %}

<a href="https://combine.org.au/" title="COMBINE">
  <img alt="COMBINE" src="https://combineau.files.wordpress.com/2016/03/combine_logo_2016_draft_v5.png" height="200" />
</a>

<h2 id="general">General Information</h2>

<!--
  INTRODUCTION

  Edit the general explanatory paragraph below if you want to change
  the pitch.
-->
<p>
In this workshop, you will be learning how to analyse RNA-seq count data, using
R. This will include reading the data into R, quality control and performing
differential expression analysis and gene set testing, with a focus on the
limma-voom analysis workflow. You will learn how to generate common plots for
analysis and visualisation of gene expression data, such as boxplots and heatmaps.
You will also be learning how alignment and counting of raw RNA-seq data can be
performed in R. In addition, you will be introduced to Degust, a user-friendly
interactive tool for RNA-seq analysis.  This workshop is aimed at biologists
interested in learning how to perform differential expression analysis of
RNA-seq data when reference genomes are available.
</p>

<p>
With special guest speakers
<strong><a href="https://en.wikipedia.org/wiki/Alicia_Oshlack">Alicia Oshlack</a></strong> and
<strong><a href="http://bioinf.wehi.edu.au/folders/shi/Wei_Shi.html">Wei Shi</a></strong>.
</p>
<!--
  AUDIENCE

  Explain who your audience is.  (In particular, tell readers if the
  workshop is only open to people from a particular institution.
-->
<p id="who">
  <strong>Who:</strong>
  The course is aimed at graduate students and other researchers.
  <strong>Some basic R knowledge is assumed</strong>.
  This is not an introduction to R course.
  If you are not familiar with the R statistical programming language we
  strongly encourage you to work through an introductory R course before you
  attend this workshop. We recommend the Software Carpentry
  <a href="http://swcarpentry.github.io/r-novice-gapminder/">R for Reproducible Scientific Analysis</a>
  lessons up to and including vectorisation (topic 9).
</p>

<!--
  LOCATION

  This block displays the address and links to maps showing directions
  if the latitude and longitude of the workshop have been set.  You
  can use http://itouchmap.com/latlong.html to find the lat/long of an
  address.
-->
{% if page.latlng %}
<p id="where">
  <strong>Where:</strong>
  {{page.address}}.
  Get directions with
  <a href="//www.openstreetmap.org/?mlat={{page.latlng | replace:',','&mlon='}}&zoom=16">OpenStreetMap</a>
  or
  <a href="//maps.google.com/maps?q={{page.latlng}}">Google Maps</a>.
</p>
{% endif %}

<p>
Thank you to VLSCI for hosting this workshop.
</p>
<p>
<a href="http://www.vlsci.org.au/">
    <img border="1" alt="VLSCI" src="https://combineau.files.wordpress.com/2015/05/logo_vlsci_rgb.jpg" height="80"></img></a>
<a href="https://www.embl-abr.org.au/">
    <img border="1" alt="EMBL Australian Bioinformatics Resource" src="https://www.embl-abr.org.au/wp-content/uploads/2016/03/logo.png" height="100"></img></a>
</p>


<!--
  SPECIAL REQUIREMENTS

  Modify the block below if there are any special requirements.
-->
<p id="requirements">
  <strong>Requirements:</strong> Participants must bring a laptop with a
  Mac, Linux, or Windows operating system (not a tablet, Chromebook, etc.) that they have administrative privileges
  on. They should have a few specific software packages installed (listed
  <a href="#setup">below</a>).
  If you choose to bring your own RNAseq data, it must be count data
  (because we don't have the time or computational resources to map large data
  sets during the workshop).
  If you're not sure, here is an
  <a href="http://www.bioconductor.org/help/workflows/rnaseqGene/#starting-from-count-matrices">example of count data format</a>.
</p>

<!--
  ACCESSIBILITY

  Modify the block below if there are any barriers to accessibility or
  special instructions.
-->
<p id="accessibility">
  <strong>Accessibility:</strong> We are committed to making this workshop
  accessible to everybody.
  The workshop organisers have checked that:
</p>
<ul>
  <li>The room is wheelchair / scooter accessible.</li>
  <li>Accessible restrooms are available.</li>
</ul>

<!--
  CONTACT EMAIL ADDRESS

  Display the contact email address set in the header.  If an address
  isn't set in the header, the Software Carpentry admin address is
  used.
-->
<p id="contact">
  <strong>Contact</strong>:
  Please mail
  {% if page.contact %}
    <a href='mailto:{{page.contact}}'>{{page.contact}}</a>
  {% else %}
    <a href='mailto:{{site.contact}}'>{{site.contact}}</a>
  {% endif %}
  for more information.
</p>

<hr/>

<!--
  SCHEDULE

  Show the workshop's schedule.  Edit the items and times in the table
  to match your plans.  You may also want to change 'Day 1' and 'Day
  2' to be actual dates or days of the week.
-->
<h2 id="schedule">Schedule</h2>

<!-- DO NOT EDIT SURVEY LINKS -->
<!--
<p><em>Surveys</em></p>
<p>Please be sure to complete the survey after the workshop. Link to be provided later.</p>
-->
<!--
<p><a href='{{ site.swc_pre_survey }}{{ site.github.project_title }}'>Pre-workshop Survey</a></p>
<p><a href='{{ site.swc_post_survey }}{{ site.github.project_title }}'>Post-workshop Survey</a></p>
-->

<p>
</p>

<p>
Note: this is a preliminary schedule. There may be changes to the timing and content.
</p>
<p>
If you have any trouble installing the software or packages, please arrive
at 8:45am on the first day so we can help before the workshop starts.
</p>
<p>
In the first session of Day 1 we will cover some R skills that are particularly
useful for RNAseq analysis, for example data frame manipulation, factors and
subsetting using logical statements. If you are a more advanced R user, you may
not need to attend this session.
</p>

<div class="row">
  <div class="col-md-6">
    <h3>Day 1</h3>
    <table class="table table-striped">
      <tr> <td>09:00</td>  <td>R for RNAseq</td> </tr>
      <tr> <td>11:00</td>  <td><i>Morning tea</i></td> </tr>
      <tr> <td>11:30</td>  <td>Introduction to RNA-seq theory</td> </tr>
      <tr> <td>12:00</td>  <td>Differential expression</td> </tr>
      <tr> <td>13:00</td>  <td><i>Lunch</i></td> </tr>
      <tr> <td>13:45</td>  <td>Differential expression continued</td> </tr>
      <tr> <td>15:00</td>  <td><i>Afternoon tea</i></td> </tr>
      <tr> <td>15:30</td>  <td>Differential expression continued</td> </tr>
      <tr> <td>16:30</td>  <td><i>Wrap-up</i></td> </tr>
    </table>
  </div>
  <div class="col-md-6">
    <h3>Day 2</h3>
    <table class="table table-striped">
      <tr> <td>09:30</td>  <td>Alignment and feature counting</td> </tr>
      <tr> <td>11:00</td>  <td><i>Morning tea</i></td> </tr>
      <tr> <td>11:30</td>  <td>Gene set testing</td> </tr>
      <tr> <td>13:00</td>  <td><i>Lunch</i></td> </tr>
      <tr> <td>13:45</td>  <td>Applying RNAseq (or BYO data)</td> </tr>
      <tr> <td>15:00</td>  <td><i>Afternoon tea</i></td> </tr>
      <tr> <td>15:30</td>  <td>Applying RNAseq (or BYO data) continued</td> </tr>
      <tr> <td>16:30</td>  <td><i>Wrap-up</i></td> </tr>
    </table>
  </div>
</div>

<!--
  ETHERPAD

  At `_misc/etherpad.txt` you will find a template for the etherpad.

  Display the Etherpad for the workshop.  You can set this up in
  advance or on the first day; either way, make sure you push changes
  to GitHub after you have its URL.  To create an Etherpad, go to

      http://pad.software-carpentry.org/YYYY-MM-DD-site

  where 'YYYY-MM-DD-site' is the identifier for your workshop,
  e.g., '2015-06-10-esu'.
-->
{% if page.etherpad %}
<p id="etherpad">
  <strong>Etherpad:</strong> <a href="{{page.etherpad}}">{{page.etherpad}}</a>.
  <br/>
  We will use this Etherpad for chatting, taking notes, and sharing URLs and bits of code.
</p>
{% endif %}

<hr/>

<!--
  SYLLABUS

  Show what topics will be covered.

  1. If your workshop is R rather than Python, remove the comment
     around that section and put a comment around the Python section.
  2. Some workshops will delete SQL.
  3. Please make sure the list of topics is synchronized with what you
     intend to teach.
  4. You may need to move the div's with class="col-md-6" around inside
     the div's with class="row" to balance the multi-column layout.

  This is one of the places where people frequently make mistakes, so
  please preview your site before committing, and make sure to run
  'tools/check' as well.
-->


<h2 id="syllabus">Syllabus</h2>


<!--
<div class="row">
  <div class="col-md-6">
    <h3 id="syllabus-shell">The Unix Shell</h3>
    <ul>
      <li>Files and directories</li>
      <li>History and tab completion</li>
      <li>Pipes and redirection</li>
      <li>Looping over files</li>
      <li>Creating and running shell scripts</li>
      <li>Finding things</li>
      <li><a href="{{site.swc_githubio}}/shell-novice/reference.html">Reference...</a></li>
    </ul>
  </div>
  <div class="col-md-6">
    <h3 id="syllabus-python">Programming in Python</h3>
    <ul>
      <li>Using libraries</li>
      <li>Working with arrays</li>
      <li>Reading and plotting data</li>
      <li>Creating and using functions</li>
      <li>Loops and conditionals</li>
      <li>Defensive programming</li>
      <li>Using Python from the command line</li>
      <li><a href="{{site.swc_githubio}}/python-novice-inflammation/reference.html">Reference...</a></li>
    </ul>
</div>
-->

You can find all of the lesson notes for this workshop
<a href="http://combine-australia.github.io/RNAseq-R/">here</a>.


  <!--
  <div class="col-md-6">
    <h3 id="syllabus-matlab">Programming in MATLAB</h3>
    <ul>
      <li>Working with arrays</li>
      <li>Reading and plotting data</li>
      <li>Creating and using functions</li>
      <li>Loops and conditionals</li>
      <li>Defensive programming</li>
      <li><a href="{{site.swc_githubio}}/matlab-novice-inflammation/reference.html">Reference...</a></li>
     </ul>
   </div>
</div>
   -->
<!--
<div class="row">
  <div class="col-md-6">
    <h3 id="syllabus-git">Version Control with Git</h3>
    <ul>
      <li>Creating a repository</li>
      <li>Recording changes to files: <code>add</code>, <code>commit</code>, ...</li>
      <li>Viewing changes: <code>status</code>, <code>diff</code>, ...</li>
      <li>Ignoring files</li>
      <li>Working on the web: <code>clone</code>, <code>pull</code>, <code>push</code>, ...</li>
      <li>Resolving conflicts</li>
      <li>Open licenses</li>
      <li>Where to host work, and why</li>
      <li><a href="{{site.swc_githubio}}/git-novice/reference.html">Reference...</a></li>
    </ul>
  </div>
  <div class="col-md-6">
    <h3 id="syllabus-sql">Managing Data with SQL</h3>
    <ul>
      <li>Reading and sorting data</li>
      <li>Filtering with <code>where</code></li>
      <li>Calculating new values on the fly</li>
      <li>Handling missing values</li>
      <li>Combining values using aggregation</li>
      <li>Combining information from multiple tables using <code>join</code></li>
      <li>Creating, modifying, and deleting data</li>
      <li>Programming with databases</li>
      <li><a href="{{site.swc_githubio}}/sql-novice-survey/reference.html">Reference...</a></li>
    </ul>
  </div>
</div>
-->
<hr/>

<!--
  SETUP

  Delete irrelevant sections from the setup instructions.  Each
  section is inside a 'div' without any classes to make the beginning
  and end easier to find.

  This is the other place where people frequently make mistakes, so
  please preview your site before committing, and make sure to run
  'tools/check' as well.
-->

<h2 id="setup">Setup</h2>

<p>
  To participate in this workshop, you will need
  access to the software described below. In addition, you will
  need an up-to-date web browser.
</p>

<p>
Even if you already have R installed, it is important that you have the latest
version because some of the packages we will be using will not work with earlier
versions of R. The latest version (available through the links below) is 3.2.5
for Windows and 3.2.4 for Mac 10.9 and above. For Mac 10.6-10.8 the latest
version is 3.2.1 (available from
<a href = "https://cran.r-project.org/bin/macosx/">here</a>)
and for that version you will also need to download this gfortran package
<a href = "https://cran.r-project.org/bin/macosx/tools/gfortran-4.2.3.pkg">gfortran package</a>).
</p>

<p>
  Software Carpentry maintains a list of common issues that occur during installation as a reference for instructors
  that may be useful on the
  <a href = "https://github.com/swcarpentry/workshop-template/wiki/Configuration-Problems-and-Solutions">Configuration Problems and Solutions wiki page</a>.
</p>

<div id="r"> <!-- Start of 'R' section. -->
  <h3>R</h3>

  <p>
    <a href="http://www.r-project.org">R</a> is a programming language
    that is especially powerful for data exploration, visualization, and
    statistical analysis. To interact with R, we use
    <a href="http://www.rstudio.com/">RStudio</a>.
  </p>

  <div class="row">
    <div class="col-md-4">
      <h4 id="r-windows">Windows</h4>
      <a href="https://www.youtube.com/watch?v=q0PjTAylwoU">Video Tutorial</a>
      <p>
        Install R by downloading and running
        <a href="http://cran.r-project.org/bin/windows/base/release.htm">this .exe file</a>
        from <a href="http://cran.r-project.org/index.html">CRAN</a>.
        Also, please install the
        <a href="http://www.rstudio.com/ide/download/desktop">RStudio IDE</a>.
      </p>
    </div>
    <div class="col-md-4">
      <h4 id="r-macosx">Mac OS X</h4>
      <a href="https://www.youtube.com/watch?v=5-ly3kyxwEg">Video Tutorial</a>
      <p>
        Install R by downloading and running
        <a href="http://cran.r-project.org/bin/macosx/R-latest.pkg">this .pkg file</a>
        from <a href="http://cran.r-project.org/index.html">CRAN</a>.
        Also, please install the
        <a href="http://www.rstudio.com/ide/download/desktop">RStudio IDE</a>.
      </p>
    </div>
    <div class="col-md-4">
      <h4 id="r-linux">Linux</h4>
      <p>
        You can download the binary files for your distribution
        from <a href="http://cran.r-project.org/index.html">CRAN</a>. Or
        you can use your package manager (e.g. for Debian/Ubuntu
        run <code>sudo apt-get install r-base</code> and for Fedora run
        <code>sudo yum install R</code>).  Also, please install the
        <a href="http://www.rstudio.com/ide/download/desktop">RStudio IDE</a>.
      </p>
    </div>
  </div>
  <h4>R packages</h4>
  <p>
      You will also need to install the following R packages:
      <strong>limma, edgeR, gplots, org.Mm.eg.db, EDASeq, RColorBrewer, GO.db, BiasedUrn, DESeq2, Glimma, Rsubread</strong>.
  </p>
  <p>
      These can all be obtained from <a href="https://www.bioconductor.org/">Bioconductor</a>, except for Glimma (see below).
  </p>
  <p>
      Open RStudio and run the following commands to install packages from Bioconductor:
  </p>
  <p>
      <code>> source("http://bioconductor.org/biocLite.R")</code>
  </p>
  <p>
      <code>> biocLite("limma")</code>
  </p>
  <p>
      Repeat this for each package.
  </p>
  <p>
      <code>> biocLite("edgeR")</code>
  </p>
  <p>
      <code>…</code>
  </p>
  <p>
      <code>> biocLite("RColorBrewer")</code>
  </p>

  <p>
      We will install Glimma from source
      (Mac users: Glimma_0.99.5.tgz, Windows: Glimma_0.99.5.zip).
      We can help with this during the workshop.
  </p>
  <p>
      <strong>Rsubread</strong>
  </p>
  <p>
      Mac and Linux users can install Rsubread using
  </p>
  <p>
      <code>> biocLite("Rsubread")</code>
  </p>
  <p>
      Windows users will need to set up a GVL instance. Instructions to follow.
  </p>

  <p>
  Common problems:
  <ul>
    <li>If the source command doesn't work, try https instead of http.</li>
    <li>Make sure you are directly connected to the internet and not using a proxy.</li>
    <li>Make sure the package name is in quotes.</li>
  </ul>
  If you have any trouble please see the
  <a href="https://www.bioconductor.org/install/">Instructions for installing Bioconductor packages</a>.
  You can also arrive early on the first day of the workshop for help with set up.
  </p>
</div> <!-- End of 'R' section. -->
